internal/compiler/object_tree.rs

```diff
@@ -2044,6 +2044,15 @@ pub fn visit_all_named_references(
         &Weak::new(),
         &mut |elem, parent_compo| {
             visit_all_named_references_in_element(elem, |nr| vis(nr));
+            if elem.borrow().repeated.is_some() {
+                if let ElementType::Component(base) = &elem.borrow().base_type {
+                    if base.parent_element.upgrade().is_some() {
+                        base.init_code.borrow_mut().iter_mut().for_each(|init_expr| {
+                            visit_named_references_in_expression(init_expr, vis)
+                        });
+                    }
+                }
+            }
             let compo = elem.borrow().enclosing_component.clone();
             if !Weak::ptr_eq(parent_compo, &compo) {
                 let compo = compo.upgrade().unwrap();
```

Comment:
I think the right thing to do here, is to visit the init_code here!
(instead of where you added it AND at the end of this function)

Reply:
Thanks! That's much cleaner indeed. I missed that.